From dd268cb370a2051c75c08192b9f190ea36eb05b4 Mon Sep 17 00:00:00 2001
From: anonym <anonym@riseup.net>
Date: Tue, 3 Nov 2020 12:13:10 +0100
Subject: [PATCH 3/3] Don't reset encryption choice when disabling encryption
 controls.

Without this, the mail.identity.default.encryptionpolicy pref does not
work: adding a key to an account always sets its encryptionpolicy to 0,
no matter the desired default.

https://bugzilla.mozilla.org/show_bug.cgi?id=1674995
---
 comm/mail/extensions/am-e2e/am-e2e.js | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/comm/mail/extensions/am-e2e/am-e2e.js b/comm/mail/extensions/am-e2e/am-e2e.js
index 4e350a1136e..56ba623770f 100644
--- a/comm/mail/extensions/am-e2e/am-e2e.js
+++ b/comm/mail/extensions/am-e2e/am-e2e.js
@@ -451,9 +451,6 @@ function smimeSelectCert(smime_cert) {
 function enableEncryptionControls(do_enable) {
   gRequireEncrypt.disabled = !do_enable;
   gDoNotEncrypt.disabled = !do_enable;
-  if (!do_enable) {
-    gEncryptionChoices.value = 0;
-  }
   // If we have a certificate or key configured that allows encryption,
   // then we are able to encrypt drafts, too.
   gEncryptDrafts.disabled = !do_enable;
-- 
2.34.1

